pkg/aggregator/ckey/key.go

```diff
@@ -75,7 +75,7 @@ type KeyGenerator struct {
 
 // Generate returns the ContextKey hash for the given parameters.
 // tagsBuf is re-arranged in place and truncated to only contain unique tags.
```

Comment:
Instead of modifying a value in-place, would it make sense to move the sort-and-dedup functionality into HashingTagsBuilder?  Then this method (Generate) would boil down to just constant ^ murmur(name) ^ murmur(hostname) ^ HashingTagsBuilder.DedupeAndHash().

Reply:
Yes, with the latest changes it should be possible. But then it should probably be renamed and moved somewhere under aggregator, since it is becoming even more specific to context key generation. That would be a fairly large refactor, so maybe it would be better to do in a separate PR.